pkg/provisioningv2/rke2/planner/encryptionkeyrotation.go

```diff
@@ -31,46 +77,46 @@ const (
 // match the init node, and either finishing if the status is reencrypt_finished, or applying the next secrets-encrypt
 // command.
 func (p *Planner) rotateEncryptionKeys(cp *rkev1.RKEControlPlane, clusterPlan *plan.Plan) error {
+	// ensure operating on latest control plane object
```

Comment:
This breaks convention. Additionally, we will be retrieving the latest `rkeControlPlane` from the cache vs. from directly against the API server IIRC, so this isn't even guaranteed to be the latest `rkeControlPlane`. 

If the goal is to always ensure we are only operating on the latest version of the `rkeControlPlane` object, we can store/compare the `resourceVersion` of the objects, although that may pose problems when rollback/recreation of the rkecontrolplane object occurs. We can also simply retrieve the latest `rkeControlPlane` object from the api server and compare resourceVersions that way, but still, not ideal I don't think.

Reply:
In this case `rkeControlPlanes` is the `Client`, not the `Cache` so I think this should be the latest, right? Also I agree that it's not necessarily ideal, but from what I can tell there are too many unknowns to do it any other way given the current implementation.

Reply:
Yes, if `rkeControlPlanes` is the client it will be the latest; however, this is not OK for us to do and we should not introduce behavior that does this within our planner controller.

Reply:
Ok, I should be able to remove this then. Given the 2 phase system I've been able to reduce it to, running through the entirety of a phase again should effectively be a no-op because the plans are in sync.